### DISCUSSION TITLE
Songs by the same artist repeating

### DISCUSSION THREAD

**Discussion by Marcioghiraldelli** [AzuraCast/AzuraCast#7438] [General]
I have a rather annoying problem with Azuracast, which is the repetition of songs. I've tried using random and shuffled, and it still doesn't work. Sometimes the same artist plays over and over.
I don't really know how the Azuracast algorithm works, which checks and tries not to play the same artist over and over, but it doesn't seem to work well. For example, if an artist has 10 songs, the system should prevent the same artist from playing songs over and over. Do you also have this problem? I've used other software and they didn't have this problem.

In Repeat Prevention Interval (Minutes),
I set it to 60 and other times to 30, and it didn't improve much.

**Comment by paulcjkp** [comment]
If you're setting your Duplicate Prevention Time Range to 30 minutes, that might be why you're seeing the same artist repeating.  When Liquidsoap is looking for a song to queue, it will check to see if the artist has played in the last 30 minutes, and if not, it'll add it to the queue. Try increasing it by a lot more (I have mine set to 1096, which is 18 hours).
